MCDataProducts/inc/CosmicLivetime.hh

```diff
@@ -42,6 +42,10 @@ namespace mu2e {
         _livetime = _primaries / (M_PI * _area * _fluxConstant * (EfToOneMinusGamma - EiToOneMinusGamma) / (1. + eslope));
     }
 
+    CosmicLivetime( float livetime ){
```

Comment:
Please use initializer syntax and initialize all variables.

Reply:
I understand they are meaningless in this case but they should be initialized and not have values of random uninitialized memory.

Use the same values as are used in the default c'tor.  You should also initialize lifetime with initializer syntax.